libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts

```diff
@@ -205,17 +197,13 @@ function translatePreBuiltCohortFilterForDataset(
         localization.Core.PreBuiltCohort.notACategoricalFeature
       ];
     }
-    const index: number[] = [];
+    let index: number[] = [];
     const categorcialValues =
       jointDataset.metaDict[jointDatasetFeatureName].sortedCategoricalValues;
     if (categorcialValues !== undefined) {
-      for (const categoricalValue of preBuiltCohortFilter.arg) {
-        const indexCategoricalValue =
-          categorcialValues.indexOf(categoricalValue);
-        if (indexCategoricalValue !== -1) {
-          index.push(indexCategoricalValue);
-        }
-      }
+      index = preBuiltCohortFilter.arg
```

Comment:
I feel like all of these can just be refactored into one helper function.. although on the other hand I guess 3 lines of code isn't so much that it's really worth it to refactor perhaps?

Reply:
Yeah, map and filter are self explanatory constructs so may not need to define a function here.